infra/lib/common.ts

```diff
@@ -136,6 +140,30 @@ export abstract class IntegrationTestStackEnvironment<
    * The Amplify configuration description for this environment.
    */
   public config: AmplifyConfig;
+
+  /**
+   * Creates a cron-scheduled Lambda which cleans up user pool users
+   * every day.
+   * 
+   * @param userPool The user pool to clean up.
+   */
+  protected createUserCleanupJob(userPool: cognito.IUserPool) {
+    const lambda = new lambda_nodejs.NodejsFunction(this, 'cleanup-users', {
+      entry: './lib/cleanup-users.ts',
+      runtime: Runtime.NODEJS_18_X,
+      environment: {
+        USER_POOL_ID: userPool.userPoolId,
+      },
+      timeout: cdk.Duration.minutes(5),
+    });
+    userPool.grant(lambda, "cognito-idp:ListUsers", "cognito-idp:AdminDeleteUser");
+
+    const cronRule = new events.Rule(this, 'cleanup-users-rule', {
+      // Run daily at midnight
+      schedule: events.Schedule.expression('cron(0 0 ? * MON-FRI *)'),
```

Comment:
nit: Could this run every day, not just mon-fri? In the future, am thinking of scheduling some very long running, high-dimension matrix suites on weekends to limit pool load, and might as well keep this running. IIUC, the lambda will run fast/bill low when pool is empty.

Reply:
Can we revisit then if it becomes an issue?

Reply:
Sure, that's fine. We already run the integ tests daily including weekends but this should be fine for now.